smurfs/src/reducers/index.js

```diff
@@ -1,3 +1,50 @@
+import {
+  FETCHING,
+  FETCHING_SUCCESS,
+  FETCHING_FAILURE,
+  ADDING,
+  ADDED,
+  ERROR} from '../actions'
+
+const initialState = {
+  smurfs: [],
+  fetchingSmurfs: false,
+  addingSmurf: false,
+  updatingSmurf: false,
+  deletingSmurf: false,
+  error: null
+}
+
+const smurfReducer = (state = initialState, action) => {
+  switch(action.type) {
+    case FETCHING:
+      return{...state, fetchingSmurfs: true};
+
+      case FETCHING_SUCCESS:
+        return{...state,
+        fetchingSmurfs: false,
+      smurfs: [...state.smurfs, ...action.payload]
```

Comment:
I'm not sure if you need to spread the action.payload here, I'd recommend to console.log the result of this operation and see if that's what you want.